graylog2-server/src/main/java/org/graylog2/indexer/fieldtypes/utils/FieldTypeDTOsMerger.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog2.indexer.fieldtypes.utils;
+
+import org.graylog2.indexer.fieldtypes.FieldTypeDTO;
+import org.graylog2.indexer.indexset.CustomFieldMappings;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+public class FieldTypeDTOsMerger {
```

Comment:
I'd suggest to simplify that class to a utility (=static method call). You will still keep the separated logic, simple unit test and get rid of all the dependency injection, pulling it through constructors...

Reply:
I do not generally like using static methods. 
I am also aware that merging is something that can be changed (for now it was decided to use 2 latest indices and custom mappings), so I am pretty happy that this is clearly visible as a separate dependency, not just a call to a static method.